### ISSUE TITLE
Gear: UPDATE gear item from single view page

### ISSUE THREAD

**Issue by ancasimon** [ancasimon/front-end-capstone#18]
## User Story
As a user, when I click the Edit button on the Single Gear Item page, I should be redirected to the Edit Gear Item page and should see a form pre-populated with all the information for the item I am editing, where I  can make edits and, when I press the Update button, I should see my changes saved in Firebase  and I should be redirected to the single view page for that gear item.

## AC
WHEN I click Edit on an item on the Single Gear Item View page,
THEN I go to an Edit Gear Item page,
AND I see all the current values for the gear item's properties pre-populated (see ticket #11 ),
AND I can change any of them,
AND when I click Update Gear Item button on this form,
THEN the changes I made are saved,
AND I go back to the the single gear item view page where I can see the changes I just made.

## Dev Notes
* We are going to add Edit button and action on the Single View page - but we are relying on the edit feature already done for the Edit button from the main Gear page for the actual editing! So that needs to be done first!!
* In the `SingleGear.js` file:
1. Import the `Link` component from the `react-router-dom` package. 
1. Inside the `return` > inside the div > add a `Link` tag here for the Edit and give it button classes to make it look like a button. 
1. In the `render`, before the `return`, declare a variable that stores the `gearItemId`, which we can pull from params (`const { gearItemId } = this.props.match.params;`) and another variable that holds the path to the Edit page where we want to take the user when they click Edit here (` const editLink = `/edit/${itemId}`;`). 
1. Add the `to` attribute to the Link tag with the editLink object we just defined. 
1. NOTE: THIS GIVES US THE EDIT FUNCTIONALITY BUT IT RETURNS THE USER TO THE GEAR PAGE - WHILE WE WANT TO GO BACK TO THE SINGLE VIEW PAGE -SO THE NEXT STEPS MAKE THE NECESSARY CHANGES TO RETURN THE USER TO THE SINGLE VIEW PAGE:
* To be able to return to the single view page after editing: In `SingleGear.js` file: 
1. Add a variable `currentpath` with a null value to state.
1. In the `componentDidMount()` method > when setting the state of the item, also set the state for the `currentpath` object with data from props (`this.props.location.pathname`) - which saves the url of this page.
1. Inside the render, add the object `{ currentpath }` and pull its value from state. 
1. Declare a new `const` variable called `newroute` that has 2 variables: one called `pathname` (MUST BE THIS LABEL) that is set to the url where we want the user to go when clicking Edit (` `/edit/${itemId}` `), and a second one to something we called `previouspath` that takes the value from the `currentpath` object. --THIS IS what we have to do to pass the older path props in a Link react component!
1. Below > inside the `return` > inside the `div` > In the `Link` tag for the Edit button > change the `to` attribute to the `newroute` object defined above. -THIS IS WHERE WE ARE PASSING IT TO THE EDIT COMPONENT.
* In `EditGearItem.js`:
1. Define proptypes for the `previouspath` object coming in. 
1. In the `updateGearItem` function here > in the `gearData.putGear` function call > in the `then()` > get the `previouspath` object from props (`const { previouspath } = this.props.location;`) and then change the current statement telling the app where to go after saving the edit into a ternary that decides whether it should go to the Gear page or back to the individual single view page: 
```
previouspath === '/gear' ? this.props.history.push('/gear') : this.props.history.push(previouspath.currentpath);
```
* Lastly, we need to pass in the current path for items when user is coming from the Gear page - but the user is actually coming in to edit from the gear item detail in that case, so > in the `GearItem.js` file:
1. Add state and set the `currentpath` to `'/stuff'` in state.
1. Then, inside the `render` but before the `return`, declare an object that extracts the `currentpath` object value from state. 
1. Then - just like in `SingleGear.js` - Declare a new `const` variable called `newroute` that has 2 variables: one called `pathname` (MUST BE THIS LABEL) that is set to the url where we want the use to go when clicking Edit (` `/edit/${itemId}` `), and a second one to something we called `previouspath` that takes the value from the `currentpath` object. --THIS IS what we have to do to pass the older path props in a Link react component!
1. LAST: Below > inside the `return` > inside the div > In the `Link` tag for the Edit button > change the `to` attribute to the `newroute` object defined above. -THIS IS WHERE WE ARE PASSING IT TO THE EDIT COMPONENT.

**Comment by ancasimon** [comment]
<img width="674" alt="edit_gear" src="https://user-images.githubusercontent.com/59633395/84709880-a630f380-af28-11ea-95a6-b954a0496362.png">
